From c8011040bf7190c1c943f6449cfa9f8a2651ef5e Mon Sep 17 00:00:00 2001
From: Zhou Guanghui <zhouguanghui1@huawei.com>
Date: Mon, 10 Oct 2022 08:28:47 +0000
Subject: [PATCH] mm/hugetlb: support disable clear hugepage

Signed-off-by: Zhou Guanghui <zhouguanghui1@huawei.com>
---
 arch/arm64/Kconfig | 7 +++++++
 mm/hugetlb.c       | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 1a02b7aa0..3a6ab0f17 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -2108,6 +2108,13 @@ config ASCEND_SVSP
 	  Therefore, in the sva scenario, an svsp is created for each target process
 	  address space.
 
+config ASCEND_CLEAR_HUGEPAGE_DISABLE
+	bool "Disable clear hugepage"
+	default n
+	help
+	  Disable clear hugepage when alloc hugepages to improve the hugepage
+	  application performance.
+
 endif
 
 endmenu
diff --git a/mm/hugetlb.c b/mm/hugetlb.c
index 679f4037b..dc4f9ba4e 100644
--- a/mm/hugetlb.c
+++ b/mm/hugetlb.c
@@ -4836,7 +4836,9 @@ retry:
 			ret = vmf_error(PTR_ERR(page));
 			goto out;
 		}
+#ifndef CONFIG_ASCEND_CLEAR_HUGEPAGE_DISABLE
 		clear_huge_page(page, address, pages_per_huge_page(h));
+#endif
 		__SetPageUptodate(page);
 		new_page = true;
 
-- 
2.17.1

