From 1afac6b004f0ece125a1e2eedcd8db7aba694324 Mon Sep 17 00:00:00 2001
From: Chen Jun <chenjun102@huawei.com>
Date: Sat, 3 Jun 2023 13:06:44 +0800
Subject: [PATCH] acpi: Do not pass IRQF_TRIGGER_RISING

Signed-off-by: Chen Jun <chenjun102@huawei.com>
---
 drivers/acpi/dt_apei.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/acpi/dt_apei.c b/drivers/acpi/dt_apei.c
index 1e2a6319f..9bb2b4256 100644
--- a/drivers/acpi/dt_apei.c
+++ b/drivers/acpi/dt_apei.c
@@ -59,7 +59,7 @@ static int dt_apei_register_irq(struct device_node *np)
 	}
 
 	rc = request_threaded_irq(irq, NULL, handle_threaded_apei_irq,
-				IRQF_TRIGGER_RISING | IRQF_ONESHOT,
+				IRQF_ONESHOT,
 				"APEI:HED", NULL);
 	if (rc != 0) {
 		pr_err("%s: request apei irq fail (%d)\n",
-- 
2.17.1

