From 376dd2ca5085472d102b621d4a411a1b1d981c76 Mon Sep 17 00:00:00 2001
From: lansheng <lansheng@huawei.com>
Date: Sat, 16 Jul 2022 08:05:12 +0000
Subject: [PATCH 4/8] huawei add emerge reboot notifier

---
 include/linux/reboot.h | 3 +++
 kernel/reboot.c        | 9 +++++++++
 2 files changed, 12 insertions(+)

diff --git a/include/linux/reboot.h b/include/linux/reboot.h
index 3734cd8f3..acb10fa57 100644
--- a/include/linux/reboot.h
+++ b/include/linux/reboot.h
@@ -42,6 +42,9 @@ extern int reboot_force;
 
 extern int register_reboot_notifier(struct notifier_block *);
 extern int unregister_reboot_notifier(struct notifier_block *);
+#ifdef CONFIG_EULEROS_EMERGE_REBOOT_NOTIFIER
+extern struct atomic_notifier_head emerge_notifier_list;
+#endif
 
 extern int devm_register_reboot_notifier(struct device *, struct notifier_block *);
 
diff --git a/kernel/reboot.c b/kernel/reboot.c
index af6f23d8b..f3ce6bdd0 100644
--- a/kernel/reboot.c
+++ b/kernel/reboot.c
@@ -53,6 +53,12 @@ int reboot_force;
 void (*pm_power_off_prepare)(void);
 EXPORT_SYMBOL_GPL(pm_power_off_prepare);
 
+#ifdef CONFIG_EULEROS_EMERGE_REBOOT_NOTIFIER
+ATOMIC_NOTIFIER_HEAD(emerge_notifier_list);
+/* exported here for compatibility because the struct has been used by products. */
+EXPORT_SYMBOL(emerge_notifier_list);
+#endif
+
 /**
  *	emergency_restart - reboot the system
  *
@@ -63,6 +69,9 @@ EXPORT_SYMBOL_GPL(pm_power_off_prepare);
  */
 void emergency_restart(void)
 {
+#ifdef CONFIG_EULEROS_EMERGE_REBOOT_NOTIFIER
+	atomic_notifier_call_chain(&emerge_notifier_list, 0, NULL);
+#endif
 	kmsg_dump(KMSG_DUMP_EMERG);
 	machine_emergency_restart();
 }
-- 
2.33.0

