From 66f89a55398c4961b55ae0a7bdccd85fc3b414e2 Mon Sep 17 00:00:00 2001
From: lansheng <lansheng@huawei.com>
Date: Sat, 16 Jul 2022 08:19:50 +0000
Subject: [PATCH 6/8] huawei kbox export security add hooks

---
 security/security.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/security/security.c b/security/security.c
index 4fb58543e..b068d077a 100644
--- a/security/security.c
+++ b/security/security.c
@@ -474,8 +474,13 @@ static int lsm_append(const char *new, char **result)
  *
  * Each LSM has to register its hooks with the infrastructure.
  */
+#ifdef CONFIG_EULEROS_KBOX_EXPORT_SECURITY_HOOKS
+void security_add_hooks(struct security_hook_list *hooks, int count,
+		char *lsm)
+#else
 void __init security_add_hooks(struct security_hook_list *hooks, int count,
 				char *lsm)
+#endif
 {
 	int i;
 
-- 
2.33.0

