diff --git a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
old mode 100644
new mode 100755
index 7db9eab6..972b8810
--- a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
@@ -52,6 +52,7 @@
 #ifdef ROCE_BONDING
 #include "hns_roce_bond.h"
 #endif
+int g_roce_port[MAX_DEVICE_ID] = {0};
 
 enum {
 	CMD_RST_PRC_OTHERS,
@@ -95,6 +96,40 @@ static const struct {
 	  HNS_ROCE_CMD_READ_SCCC_BT0, HNS_ROCE_CMD_WRITE_SCCC_BT0 },
 };
 
+int hns_roce_v2_set_roce_port(int dev_id, int roce_port)
+{
+    if (dev_id < 0 || dev_id >= MAX_DEVICE_ID) {
+        pr_err("hns3: Device id out of range.(dev_id = %d)\n", dev_id);
+        return -EINVAL;
+    }
+    if (roce_port == 0) {
+        g_roce_port[dev_id] |= UDP_PORT_AUTO;
+    } else {
+        g_roce_port[dev_id] = roce_port;
+    }
+    return 0;
+}
+EXPORT_SYMBOL(hns_roce_v2_set_roce_port);
+
+int hns_roce_v2_get_roce_port(int dev_id, int *roce_port)
+{
+    if (roce_port == NULL) {
+        pr_err("hns3: Invalid param, roce_port is NULL.\n");
+        return -EINVAL;
+    }
+    if (dev_id < 0 || dev_id >= MAX_DEVICE_ID) {
+        pr_err("hns3: Device id out of range.(dev_id = %d)\n", dev_id);
+        return -EINVAL;
+    }
+    if (g_roce_port[dev_id] == 0) {
+        *roce_port = DEFAULT_UDP_PORT;
+    } else {
+        *roce_port = g_roce_port[dev_id];
+    }
+    return 0;
+}
+EXPORT_SYMBOL(hns_roce_v2_get_roce_port);
+
 static void set_data_seg_v2(struct hns_roce_v2_wqe_data_seg *dseg,
 			    struct ib_sge *sg, struct hns_roce_dev *hr_dev)
 {
@@ -4843,6 +4878,27 @@ static int fill_cong_field(struct ib_qp *ibqp, const struct ib_qp_attr *attr,
 	return 0;
 }
 
+static int hns_roce_get_die_id(struct hns_roce_dev *hr_dev, u32 *die_id)
+{
+	struct hns_roce_v2_priv *priv = hr_dev->priv;
+	struct hnae3_handle *handle = priv->handle;
+	struct ib_device *ibdev = &hr_dev->ib_dev;
+
+	if (die_id == NULL) {
+		ibdev_err(ibdev, "Get device info err, die_id is null.\n.");
+		return -EINVAL;
+	}
+
+	if (handle->ae_algo->ops->get_device_info == NULL) {
+		ibdev_err(ibdev, "Get device info err, not support.\n.");
+		return -ENOTSUPP;
+	}
+
+	handle->ae_algo->ops->get_device_info(handle, NULL, NULL, NULL, die_id);
+
+	return 0;
+}
+
 static int hns_roce_v2_set_path(struct ib_qp *ibqp,
 				const struct ib_qp_attr *attr,
 				int attr_mask,
@@ -4857,6 +4913,8 @@ static int hns_roce_v2_set_path(struct ib_qp *ibqp,
 	int is_roce_protocol;
 	u16 vlan_id = 0xffff;
 	bool is_udp = false;
+	u32 die_id;
+	int udp_port;
 	u8 ib_port;
 	u8 hr_port;
 	int ret;
@@ -4875,6 +4933,7 @@ static int hns_roce_v2_set_path(struct ib_qp *ibqp,
 		is_udp = (gid_attr->gid_type == IB_GID_TYPE_ROCE_UDP_ENCAP);
 	}
 
+
 	/* Only HIP08 needs to set the vlan_en bits in QPC */
 	if (vlan_id < VLAN_N_VID &&
 	    hr_dev->pci_dev->revision == PCI_REVISION_ID_HIP08) {
@@ -4898,10 +4957,24 @@ static int hns_roce_v2_set_path(struct ib_qp *ibqp,
 		return -EINVAL;
 	}
 
+	ret = hns_roce_get_die_id(hr_dev, &die_id);
+    	
+	if (ret != 0) {
+        	ibdev_err(ibdev, "Fail to set path, get die_id err ret:%d\n", ret);
+        	return ret;
+        }
+        if ((g_roce_port[die_id] & UDP_PORT_AUTO) != 0) {
+           	 udp_port = (hr_qp->qpn % UDP_PORT_LEN) + UDP_PORT_MIN;
+       		 g_roce_port[die_id] = udp_port;
+         	 g_roce_port[die_id] |= UDP_PORT_AUTO;
+    	} else if (g_roce_port[die_id] == 0) {
+        	udp_port = DEFAULT_UDP_PORT;
+    	} else {
+        	udp_port = g_roce_port[die_id];
+    	}
+
 	hr_reg_write(context, QPC_UDPSPN,
-		     is_udp ? rdma_get_udp_sport(grh->flow_label, ibqp->qp_num,
-						 attr->dest_qp_num) :
-				    0);
+		is_udp ? udp_port : 0);
 
 	hr_reg_clear(qpc_mask, QPC_UDPSPN);
 
@@ -5708,27 +5781,6 @@ static int hns_roce_v2_modify_cq(struct ib_cq *cq, u16 cq_count, u16 cq_period)
 	return ret;
 }
 
-static int hns_roce_get_die_id(struct hns_roce_dev *hr_dev, u32 *die_id)
-{
-	struct hns_roce_v2_priv *priv = hr_dev->priv;
-	struct hnae3_handle *handle = priv->handle;
-	struct ib_device *ibdev = &hr_dev->ib_dev;
-
-	if (die_id == NULL) {
-		ibdev_err(ibdev, "Get device info err, die_id is null.\n.");
-		return -EINVAL;
-	}
-
-	if (handle->ae_algo->ops->get_device_info == NULL) {
-		ibdev_err(ibdev, "Get device info err, not support.\n.");
-		return -ENOTSUPP;
-	}
-
-	handle->ae_algo->ops->get_device_info(handle, NULL, NULL, NULL, die_id);
-
-	return 0;
-}
-
 static void hns_roce_bbox_report(struct hns_roce_dev *hr_dev, u32 sub_id)
 {
 	hns_bbox_excep_report_fun hns_bbox_excep_report =
diff --git a/drivers/infiniband/hw/hns/hns_roce_hw_v2.h b/drivers/infiniband/hw/hns/hns_roce_hw_v2.h
index dd974a7b..4987b838 100644
--- a/drivers/infiniband/hw/hns/hns_roce_hw_v2.h
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v2.h
@@ -115,6 +115,13 @@
 
 #define HNS_ROCE_V2_TABLE_CHUNK_SIZE		(1 << 18)
 
+#define  MAX_DEVICE_ID 64
+#define  DEFAULT_UDP_PORT 0x12b7
+#define  UDP_PORT_MIN 4097
+#define  UDP_PORT_MAX 65535
+#define  UDP_PORT_LEN (UDP_PORT_MAX - UDP_PORT_MIN)
+#define  UDP_PORT_AUTO 0x10000
+#define  UDP_PORT_RANGE 0xFFFF
 enum {
 	HNS_ROCE_CMD_FLAG_IN = BIT(0),
 	HNS_ROCE_CMD_FLAG_OUT = BIT(1),
