From ed9c6bd134a46f4e57baf7f08a1b5f3e777a7bc6 Mon Sep 17 00:00:00 2001
From: Chen Jun <chenjun102@huawei.com>
Date: Tue, 7 May 2019 21:59:51 -0400
Subject: [PATCH] Add secure compilation option

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index d5ec09ed8..bfbe39c4a 100755
--- a/Makefile
+++ b/Makefile
@@ -995,8 +995,8 @@ KBUILD_CPPFLAGS += $(KCPPFLAGS)
 KBUILD_AFLAGS   += $(KAFLAGS)
 KBUILD_CFLAGS   += $(KCFLAGS)
 
-KBUILD_LDFLAGS_MODULE += --build-id=sha1
-LDFLAGS_vmlinux += --build-id=sha1
+KBUILD_LDFLAGS_MODULE += --build-id=none
+LDFLAGS_vmlinux += --build-id=none
 
 KBUILD_LDFLAGS	+= -z noexecstack
 KBUILD_LDFLAGS	+= $(call ld-option,--no-warn-rwx-segments)
-- 
2.17.1

