diff --git a/drivers/infiniband/hw/hns/hns_roce_bond.c b/drivers/infiniband/hw/hns/hns_roce_bond.c
index 2ec8779..9bc36d1 100755
--- a/drivers/infiniband/hw/hns/hns_roce_bond.c
+++ b/drivers/infiniband/hw/hns/hns_roce_bond.c
@@ -441,8 +441,11 @@ void hns_roce_bond_clear_all(void)
 	int chip_id;
 
 	mutex_lock(&g_roce_env_info.env_info_mutex);
-	for (chip_id = 0; chip_id < g_roce_env_info.chip_cnt; chip_id++)
-		single_chip_clear(chip_id);
+	for (chip_id = 0; chip_id < ROCE_BOND_CHIP_MAX; chip_id++) {
+		if (g_roce_env_info.single_chip_info[chip_id].chip_init_flag) {
+			single_chip_clear(chip_id);
+		}
+	}
 
 	mutex_unlock(&g_roce_env_info.env_info_mutex);
 }
diff --git a/drivers/infiniband/hw/hns/hns_roce_hem.c b/drivers/infiniband/hw/hns/hns_roce_hem.c
index 572f60a..2261708 100644
--- a/drivers/infiniband/hw/hns/hns_roce_hem.c
+++ b/drivers/infiniband/hw/hns/hns_roce_hem.c
@@ -50,13 +50,20 @@ void *hns_roce_mem_alloc(size_t size, ALLOC_TYPE alloc_type, u32 num, gfp_t flag
 {
 	hns3_numa_mem_alloc_fun hns3_numa_mem_alloc =
 		(hns3_numa_mem_alloc_fun)(uintptr_t)symbol_get(hns3_numa_mem_alloc);
+	void *addr = NULL;
 
 	if (hns3_numa_mem_alloc == NULL) {
 		pr_err("hns3: hns3_numa_mem_alloc is NULL!\n");
 		return NULL;
 	}
 
-	return hns3_numa_mem_alloc(size, alloc_type, num, flags);
+	addr = hns3_numa_mem_alloc(size, alloc_type, num, flags);
+
+	if (hns3_numa_mem_alloc != NULL) {
+		symbol_put(hns3_numa_mem_alloc);
+	}
+
+	return addr;
 }
 
 void *hns_roce_dma_alloc_coherent(struct device *dev, size_t size,
@@ -64,13 +71,20 @@ void *hns_roce_dma_alloc_coherent(struct device *dev, size_t size,
 {
 	hns3_numa_dma_alloc_coherent_fun hns3_numa_dma_alloc_coherent =
 		(hns3_numa_dma_alloc_coherent_fun)(uintptr_t)symbol_get(hns3_numa_dma_alloc_coherent);
+	void *addr = NULL;
 
 	if (hns3_numa_dma_alloc_coherent == NULL) {
 		dev_err(dev, "hns3_numa_dma_alloc_coherent is NULL!\n");
 		return NULL;
 	}
 
-	return hns3_numa_dma_alloc_coherent(dev, size, dma_handle, flags);
+	addr = hns3_numa_dma_alloc_coherent(dev, size, dma_handle, flags);
+
+	if (hns3_numa_dma_alloc_coherent != NULL) {
+		symbol_put(hns3_numa_dma_alloc_coherent);
+	}
+
+	return addr;
 }
 
 void hns_roce_dma_free_coherent(struct device *dev, size_t size,
@@ -84,7 +98,11 @@ void hns_roce_dma_free_coherent(struct device *dev, size_t size,
 		return;
 	}
 
-	return hns3_numa_dma_free_coherent(dev, size, mem, dma_addr);
+	hns3_numa_dma_free_coherent(dev, size, mem, dma_addr);
+
+	if (hns3_numa_dma_free_coherent != NULL) {
+		symbol_put(hns3_numa_dma_free_coherent);
+	}
 }
 #endif
 
diff --git a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
index 390952f..7ffab04 100644
--- a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
@@ -5760,6 +5760,14 @@ static void hns_roce_bbox_report(struct hns_roce_dev *hr_dev, u32 sub_id)
 	}
 
 	hns_bbox_excep_report(die_id, except_id);
+
+	if (hns_bbox_excep_report != NULL) {
+		symbol_put(hns_bbox_excep_report);
+	}
+
+	if (hns_get_bbox_excepid != NULL) {
+		symbol_put(hns_get_bbox_excepid);
+	}
 }
 
 static void hns_roce_irq_work_handle(struct work_struct *work)
