From fc5f9a198bcf5988accf29717357209957543f99 Mon Sep 17 00:00:00 2001
Date: Wed, 5 Jul 2023 10:07:08 +0800
Subject: [PATCH] hns3-switch-speed-bug-fix

---
 .../ethernet/hisilicon/hns3/hns3_ethtool.c    | 25 ++++++++++++++++---
 1 file changed, 22 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/hisilicon/hns3/hns3_ethtool.c b/drivers/net/ethernet/hisilicon/hns3/hns3_ethtool.c
index 27285ef..f914847 100644
--- a/drivers/net/ethernet/hisilicon/hns3/hns3_ethtool.c
+++ b/drivers/net/ethernet/hisilicon/hns3/hns3_ethtool.c
@@ -866,10 +866,13 @@ static int hns3_check_ksettings_param(const struct net_device *netdev,
 static int hns3_set_link_ksettings(struct net_device *netdev,
 				   const struct ethtool_link_ksettings *cmd)
 {
+	static u32 phy_speed[] = { SPEED_1000, SPEED_100, SPEED_10 };
 	struct hnae3_handle *handle = hns3_get_handle(netdev);
 	struct hnae3_ae_dev *ae_dev = hns3_get_ae_dev(handle);
 	const struct hnae3_ae_ops *ops = handle->ae_algo->ops;
+	bool speed_sup_flag = false;
 	int ret;
+	int i;
 
 	/* Chip don't support this mode. */
 	if (cmd->base.speed == SPEED_1000 && cmd->base.duplex == DUPLEX_HALF)
@@ -882,9 +885,25 @@ static int hns3_set_link_ksettings(struct net_device *netdev,
 
 	/* Only support ksettings_set for netdev with phy attached for now */
 	if (netdev->phydev) {
-		if (cmd->base.speed == SPEED_1000 &&
-		    cmd->base.autoneg == AUTONEG_DISABLE)
-			return -EINVAL;
+		if (!(handle->flags & HNAE3_SUPPORT_PLATFORM_DEV)) {
+			if (cmd->base.speed == SPEED_1000 &&
+				cmd->base.autoneg == AUTONEG_DISABLE)
+				return -EINVAL;
+		} else {
+			/* PHY device supports rate configuration when auto-negotiation is enabled,
+			 * and the chip does not support half-duplex.
+			 */
+			int mode_num = sizeof(phy_speed) / sizeof(phy_speed[0]);
+			for (i = 0; i < mode_num; i++) {
+				if (cmd->base.speed == phy_speed[i]) {
+					speed_sup_flag = true;
+					break;
+				}
+			}
+
+			if (cmd->base.duplex != DUPLEX_FULL || speed_sup_flag != true)
+				return -EINVAL;
+		}
 
 		return phy_ethtool_ksettings_set(netdev->phydev, cmd);
 	} else if (test_bit(HNAE3_DEV_SUPPORT_PHY_IMP_B, ae_dev->caps) &&
-- 
2.27.0

